horovod/tensorflow/xla_mpi_ops.cc

```diff
@@ -563,13 +616,18 @@ void CallbackHVDAllreduceDone(CUstream stream, void** /*buffers*/,
   VLOG(2) << "hvd-allreduce-done - End";
 }
 
+#if HAVE_CUDA
 XLA_REGISTER_CUSTOM_CALL_TARGET(CallbackHVDAllreduce, "CUDA");
 XLA_REGISTER_CUSTOM_CALL_TARGET(CallbackHVDAllreduceDone, "CUDA");
+#elif HAVE_ROCM
+XLA_REGISTER_CUSTOM_CALL_TARGET(CallbackHVDAllreduce, "ROCM");
+XLA_REGISTER_CUSTOM_CALL_TARGET(CallbackHVDAllreduceDone, "ROCM");
+#endif
 
 } // namespace
 } // namespace tensorflow
 } // namespace horovod
 
 #endif // TENSORFLOW_VERSION >= 2006000000
-#endif // HAVE_CUDA
+//#endif // HAVE_CUDA
```

Comment:
Awesome, that sounds great, and thanks again for the PR!

Reply:
> Thanks for the PR! Overall, this looks great. A couple high level comments:
> 
> * It seems we've introduced a lot of if-else statements that were previously encapsulated behind interfaces. My take is that if we're going to unify the implementations, we'll want to try and avoid these if-else branches as much as possible and just use defines, like the ones implemented in common.h [here](https://github.com/horovod/horovod/blob/master/horovod/common/common.h#L31). Curious to get your thoughts on this.
> * It would be great if we could setup some tests as part of this refactor to ensure that as we continue to add new functionality we don't break any of the ROCm code paths. At the least, a compilation test with a ROCm Docker image would be a great start.
> 
> Let me know what you think, happy to discuss some of these points in more detail if it's helpful.

Hi Travis, first of all, thank you for your feedbacks. I have removed the if-else branches in the last commit. Could you please take a look at my new changes? I added gpu_common.h in the horovod/common/ops directory for convenience since the header file, gpu_common.h, would be included by other GPU source files. I think it is better to pull it out from common.h. What is your opinion on this? I will talk to my colleagues to see what the best way is to take care of the second bullet point. If possible, can you please help with the merge?

Reply:
@tgaddair Would you please let me know if there is anything I can do to improve this PR? Thank you.